src/coreclr/src/vm/tailcallhelp.cpp

```diff
@@ -107,227 +107,20 @@ struct TailCallInfo
     }
 };
 
-static MethodDesc* s_tailCallDispatcherMD;
 MethodDesc* TailCallHelp::GetTailCallDispatcherMD()
 {
-    LIMITED_METHOD_CONTRACT;
+    CONTRACTL
+    {
+        THROWS;
+        GC_TRIGGERS;
+        INJECT_FAULT(ThrowOutOfMemory());
+    }
+    CONTRACTL_END;
```

Comment:
This function is used by the debugger:
https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/coreclr/src/debug/ee/controller.cpp#L5662-L5668

Is this contract change ok?

Reply:
This can cause type loader to run in deep inside debugger that won't fly well. 

I think you should keep the current structure where the `MethodDesc*` is cached in `s_tailCallDispatcherMD` during JITing and the debugger gets this cached copy.